cmds/modules/noded/main.go

```diff
@@ -193,6 +194,17 @@ func action(cli *cli.Context) error {
 		return hypervisor, nil
 	})
 
+	log.Info().Msg("Start Perf scheduler")
+	performanceMonitor := perf.NewPerformanceMonitor("/var/run/redis.sock")
+	performanceMonitor.InitScheduler()
+	err = performanceMonitor.RunScheduler(ctx)
+	if err != nil {
+		log.Error().Err(err).Msg("fails in scheduler")
+	}
+	bus.WithHandler("zos.perf.get", func(ctx context.Context, payload []byte) (interface{}, error) {
+		return performanceMonitor.Get(ctx, payload)
```

Comment:
handling of user input is part of the `public` API and then it should handled here not in the performanceMonitor library or object. 

This breaks `separation of concerns` and `isolation` of your design.